IronPython3Extension/Extension.cs

```diff
@@ -44,4 +119,35 @@ public void Startup(StartupParams sp)
 
         }
     }
+    internal class IsolatedPythonContext : AssemblyLoadContext
```

Comment:
This context seems similar to the one in `DSIronPython`. If so, would it make sense to move this class to DynamoCore which can be accessed in both places to avoid code duplication?

Reply:
yep, it's completely the same, it would definitely be nice to share the implementation of a few parts of this approach in the future.

But again I don't want this fix to be dependent on any changes in Dynamo for this release cycle. Said another way, my goal is that we can easily roll this fix out for IronPython and IronPython3 packages and just tell users to go get it, without having them consider what version of Dynamo 3.x they are on - so it can't depend on changes in Dynamo. Not all integrators will be necessarily grabbing our latest releases.

I'm also not sure where this type would go in DynamoCore - maybe DynamoServices makes sense along with the PythonServices stuff.

Hopefully we find a way to holistically allow packages to opt into using assembly load contexts so devs don't need to implement their own ALC.

I think we already have tasks for the holistic approach - I will file a task for at least trying to share implementation between these two packages somehow. Introducing a shared package dependent seems idiomatic in other ecosystems but I'm wary of it in Dynamo...

Reply:
filed https://jira.autodesk.com/browse/DYN-6699 to look into code sharing with a caveat about assembly load contexts.